definitions/output/all/backfill_summary_requests.js

```diff
@@ -0,0 +1,239 @@
+    'requests_backfill'
+  ]).dependencies([
+    i === 0 ? '' : `backfill_summary_requests ${iterations[i - 1].date} ${iterations[i - 1].client}`
+  ]).queries(ctx => `
+DELETE FROM ${ctx.resolve('all', 'requests')}
+WHERE date = '${iteration.date}' AND client = '${iteration.client}';
+
+CREATE TEMP FUNCTION get_ext_from_url(url STRING)
+RETURNS STRING
+LANGUAGE js
+AS """
+  try {
+    let ret_ext = url;
+
+    // Remove query parameters
+    const i_q = ret_ext.indexOf("?");
+    if (i_q > -1) {
+      ret_ext = ret_ext.substring(0, i_q);
+    }
+
+    // Get the last segment of the path after the last "/"
+    ret_ext = ret_ext.substring(ret_ext.lastIndexOf("/") + 1);
+
+    // Find the position of the last dot
+    const i_dot = ret_ext.lastIndexOf(".");
+
+    if (i_dot === -1) {
+      // No dot means no extension
+      ret_ext = "";
+    } else {
+      // Extract the extension
+      ret_ext = ret_ext.substring(i_dot + 1);
+
+      // Weed out overly long extensions
+      if (ret_ext.length > 5) {
+        ret_ext = "";
+      }
+    }
+
+    return ret_ext.toLowerCase();
+  } catch (e) {
+    return ""; // Return an empty string in case of any errors
+  }
+""";
+
+CREATE TEMP FUNCTION get_type(mime_typ STRING, ext STRING)
+RETURNS STRING
+LANGUAGE js
+AS """
+  try {
+    mime_typ = mime_typ.toLowerCase();
+
+    // Order by most unique types first
+    const uniqueTypes = ["font", "css", "image", "script", "video", "audio", "xml"];
+    for (let typ of uniqueTypes) {
+      if (mime_typ.includes(typ)) {
+        return typ;
+      }
+    }
+
+    // Special cases
+    if (mime_typ.includes("json") || ["js", "json"].includes(ext)) {
+      return "script";
+    } else if (["eot", "ttf", "woff", "woff2", "otf"].includes(ext)) {
+      return "font";
+    } else if (
+      ["png", "gif", "jpg", "jpeg", "webp", "ico", "svg", "avif", "jxl", "heic", "heif"].includes(ext)
+    ) {
+      return "image";
+    } else if (ext === "css") {
+      return "css";
+    } else if (ext === "xml") {
+      return "xml";
+    } else if (
+      ["mp4", "webm", "ts", "m4v", "m4s", "mov", "ogv", "swf", "f4v", "flv"].includes(ext) ||
+      ["flash", "webm", "mp4", "flv"].some(typ => mime_typ.includes(typ))
+    ) {
+      return "video";
+    } else if (mime_typ.includes("wasm") || ext === "wasm") {
+      return "wasm";
+    } else if (mime_typ.includes("html") || ["html", "htm"].includes(ext)) {
+      return "html";
+    } else if (mime_typ.includes("text")) {
+      // Put "text" last because it is often misused, so extension should take precedence.
+      return "text";
+    } else {
+      return "other";
+    }
+  } catch (e) {
+    return "other"; // Return "other" if there's any error
+  }
+""";
+
+CREATE TEMP FUNCTION parse_headers(headers STRING)
+RETURNS ARRAY<STRUCT<name STRING, value STRING>>
+LANGUAGE js
+AS """
+  try {
+    return headers.split(', ').map(header => {
+      const [name, value] = header.split(' = ');
```

Comment:
Did anybody have a successful attempt parsing req/respOtherHeaders?
Using existing separators gives a mess.

Reply:
Will do the best effort and skip failed headers.